### DISCUSSION TITLE
Struggling with traffic generation and visualization

### DISCUSSION THREAD

**Discussion by phdavis1027** [omnetpp/omnetpp#1138] [Q&A]
I am currently trying to create an omnetpp.ini file that iterates overall hosts and sends a trivial amount of bytes to a randomly chosen host. All sending connections close at 30s. However, the event log is report strange things like events that happen at t=4334. Also, sending apps closing with 70-90 sessions open. Here is the basic structure of the traffic generation in the file:

```
**.host[0].app[0].typename = "TcpBasicClientApp"
**.host[0].app[0].tOpen = 0s
**.host[0].app[0].tSend = 0.5s
**.host[0].app[0].tClose = 30s
**.host[0].app[0].sendBytes = 30B
**.host[0].app[0].connectAddress = /* random host */
**.host[0].numApps = 2
**.host[0].app[0].thinkTime = 0.1s
**.host[0].app[0].idleInterval = 1s
**.host[0].app[1].typename = "TcpSinkApp"
```

Additionally, I would likely to visualize my network to sanity check when things like this happen. Do I have options to do this under cmdenv?

**Comment by phdavis1027** [comment]
As an update, I changed the structure to use TcpGenericServerApp instead of TcpSinkApp, and told the client to close after 5 seconds. The simulation has been running for hours on this trivial workload and has generated almost 300gb of results. Is this normal? The event log says simulation time is still only 9.2 seconds, but that still seems suspect to me since no client should have sent requests after 5 seconds (if my reading of the docs is right, TcpGenericServerApp should time out after 2 seconds of inactivity). 

Basically, it strikes me as an improvement that my simulation isn't skipping forward hours at a time in a simtime, but it still seems suspicious to me that such seemingly simple code produces such strange results.

**Comment by levy** [comment]
If you think you have found a bug, then please try to narrow down the simulation to the simplest NED and INI files and submit an issue.

**Comment by phdavis1027** [comment]
It's possible I found a bug, but I think it's more likely I just did something wrong. I really don't know what is expected behavior for a simulation of my size and what is evidence that something is wrong inside of Omnet. For reference, there are 64 leaf hosts in this network, each sending a trivial load to one other host. It has now been running for 22 hours and has produced almost a terabyte of data.

**Reply by levy** [reply]
Writing 750MB per minute is expected.

**Comment by phdavis1027** [comment]
What is the expected runtime for a fairly uncomplicated network with 64
hosts?

Le ven. 29 mars 2024 à 04:20, Levente Mészáros ***@***.***> a
écrit :

**Reply by levy** [reply]
There's no such thing, it depends on sim-time-limit and many other factors.
